test/javascripts/turbolinks_test.coffee

```diff
@@ -1,210 +1,946 @@
-        @server.respond()
-
-        assert your_callback.calledOnce
-
-      it 'script tags are evaluated when they are the subject of a partial replace', ->
-        window.globalStub = stub()
-        @server.respondWith([200, { "Content-Type": "text/html" }, script_response]);
-
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-        assert globalStub.calledOnce
-
-      it 'script tags are not evaluated if they have [data-turbolinks-eval="false"]', ->
-        window.globalStub = stub()
-        @server.respondWith([200, { "Content-Type": "text/html" }, script_response_turbolinks_eval_false]);
-
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-        assert.equal 0, globalStub.callCount
-
-      it 'triggers the page:load event with a list of nodes that are new (freshly replaced)', ->
-        $(document).one 'page:load', (event) ->
-          ev = event.originalEvent
-          assert.equal true, ev.data instanceof Array
-          assert.equal 1, ev.data.length
-          node = ev.data[0]
-
-          assert.equal "turbo-area", node.id
-          assert.equal "turbo-area", node.getAttribute('refresh')
-
-        @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
-
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-
-        $(document).off 'page:load'
-
-      it 'does not trigger the page:before-partial-replace event more than once', ->
-        handler = stub()
-        $(document).on 'page:before-partial-replace', ->
-          handler()
-          assert handler.calledOnce
-
-        @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
-
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
-
-        $(document).off 'page:before-partial-replace'
-
-      it 'replaces and passes through the outermost nodes if a series of nodes got replaced', ->
-        currentBody = """
-          <div id="div1" refresh="div1">
-            <div id="div2" refresh-always>Refresh-always</div>
-          </div>
-        """
-
-        $("body").append($(currentBody))
-
+        assert.equal('/noScriptsOrLinkInHead', event.originalEvent.data)
+        assert.equal(0, server.requests.length)
+        done()
+
+      visit(url: 'noScriptsOrLinkInHead', options: { partialReplace: true, onlyKeys: ['turbo-area'] })
+
+    it 'supports passing request headers', (done) ->
+      visit url: 'noScriptsOrLinkInHead', options: { headers: {'foo': 'bar', 'fizz': 'buzz'} }, ->
+        assert.equal('bar', server.requests[0].requestHeaders['foo'])
+        assert.equal('buzz', server.requests[0].requestHeaders['fizz'])
+        done()
+
+    it 'calls a user-supplied callback', (done) ->
```

Comment:
What is this testing?  I guess that `page:load` is called after a successful TG navigation?  Otherwise it's just verifying that our `visit` helper is working.

The test name should give me a better clue.  I'm sorely lacking in that area.